x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/promql/PromqlLogicalPlanOptimizerTests.java

```diff
@@ -346,6 +347,47 @@ avg by (pod) (rate(network.bytes_in[1h]))
         var plan = planPromql(testQuery);
     }
 
+    /**
+     * Expect the logical plan structure:
+     * Project
+     * \_Eval
+     *   \_Limit
+     *     \_Aggregate
+     *       \_Eval
+     *         \_TimeSeriesAggregate[[...],[SUM(...,PT10M,...), COUNT(...,PT10M,...), ...], BUCKET(@timestamp,PT5M)]
+     */
+    public void testRangeSelectorWithDifferentStep() {
+        var plan = planPromql("""
+            PROMQL index=k8s step=5m sum by (pod) (avg_over_time(events_received[10m]))
+            """);
+
+        var project = as(plan, Project.class);
+        var evalOuter = as(project.child(), Eval.class);
+        var limit = as(evalOuter.child(), Limit.class);
+        var aggregate = as(limit.child(), Aggregate.class);
+        var evalMiddle = as(aggregate.child(), Eval.class);
+        var tsAggregate = as(evalMiddle.child(), TimeSeriesAggregate.class);
```

Comment:
I think we care less a bout the exact structure in this test, so we could simplify to something like

```java
var tsAggregate = plan.collect(TimeSeriesAggregate.class).getFirst();
```